modules/blockbuilder/blockbuilder_test.go

```diff
@@ -78,6 +78,37 @@ func TestBlockbuilder_lookbackOnNoCommit(t *testing.T) {
 	requireLastCommitEquals(t, ctx, client, producedRecords[len(producedRecords)-1].Offset+1)
 }
 
+func TestBlockbuilder_without_partitions_assigned(t *testing.T) {
+	ctx, cancel := context.WithCancelCause(context.Background())
+	t.Cleanup(func() { cancel(errors.New("test done")) })
+
+	k, address := testkafka.CreateCluster(t, 1, testTopic)
+
+	kafkaCommits := atomic.NewInt32(0)
+	k.ControlKey(kmsg.OffsetCommit, func(kmsg.Request) (kmsg.Response, error, bool) {
+		kafkaCommits.Inc()
+		return nil, nil, false
+	})
+
+	store := newStore(ctx, t)
+	cfg := blockbuilderConfig(t, address, []int32{})
+
+	b, err := New(cfg, test.NewTestingLogger(t), newPartitionRingReader(), &mockOverrides{}, store)
+	require.NoError(t, err)
+	require.NoError(t, services.StartAndAwaitRunning(ctx, b))
+	t.Cleanup(func() {
+		require.NoError(t, services.StopAndAwaitTerminated(ctx, b))
+	})
+
+	client := newKafkaClient(t, cfg.IngestStorageConfig.Kafka)
+	sendReq(t, ctx, client)
+
+	// Wait for record to be consumed and committed.
+	require.Eventually(t, func() bool {
```

Comment:
I think this won't test things as expected, since the atomic starts at zero it would be satisfied immediately.